File: go/pangram/pangram.go
// Pangram detect if an input contains all the letters of the english alphabet
package pangram

import (
	"slices"
	"strings"
)

// Detect if a given input is a Pangram, it contains all
// letters a-z at least once (case insensitive).
// @param input: the string to test
// @returns: true if a pangram otherwise false
func IsPangram(input string) bool {
	const alphabet string = "abcdefghijklmnopqrstuvwxyz"
	var present []string = []string{}

	for _, rune := range strings.ToLower(input) {
		char := string(rune)
		if strings.Contains(alphabet, char) && !slices.Contains(present, char) {
			present = append(present, char)
		}
	}
	return len(present) == len(alphabet)
}

File: php/pangram/Pangram.php
<?php
// Pangram exercise. See if a given string has all of the letters a-z of the english alphabet.

declare(strict_types=1);

// Check to see if the given $string uses all of the letters of the english alphabet
// @param $string: The text to check if it is a pangram or not.
// @returns: True if a pangram
function isPangram(string $string): bool
{
    $letters = array_count_values(str_split(strtolower($string)));
    for ($i = ord("a"); $i <= ord("z"); $i++) {
        $letter = chr($i);
        if (!array_key_exists($letter, $letters)) {
            return false;
        }
    }
    return true;

}
